csrc/dynamic_transform.cpp

```diff
@@ -930,21 +954,134 @@ void DynamicTransformConcretizer::concretizeReshape() {
+    // already replaced (for example one IterDomain could theoretically be
+    // input to multiple Resize ops).
+    IterDomainBuilder builder(in_id);
+
+    if (!has_const_extent) {
+      Val* new_extent = IrBuilder::create<Val>(
+          input_extent, in_id->getMaybeExpandedExtent()->dtype());
+      if (in_id->hasExpandedExtent()) {
+        builder.expanded_extent(new_extent);
+      } else {
+        builder.extent(new_extent);
+      }
+    }
+
+    if (in_id->isSymbolic()) {
+      builder.iter_type(
+          input_extent + left_expand + right_expand == 1 ? IterType::Broadcast
+                                                         : IterType::Iteration);
+    }
+
+    IterDomain* new_in_id = builder.build();
+
+    ir_utils::replaceValInExprInputs(id->definition(), orig_in_id, new_in_id);
+
+    registerConcretization(orig_in_id, new_in_id);
+    if (!orig_in_id->getMaybeExpandedExtent()->sameAs(
+            new_in_id->getMaybeExpandedExtent())) {
+      registerConcretization(
+          orig_in_id->getMaybeExpandedExtent(),
+          new_in_id->getMaybeExpandedExtent());
+    }
+  }
+
+  // Concretize each resize op using constant expand values
+  for (const auto& [id_index, concrete_resize] : info_->getResizeExtents()) {
+    const auto& [input_extent, left_expand, right_expand] = concrete_resize;
+
+    IterType iter_type = input_extent + left_expand + right_expand == 1
+        ? IterType::Broadcast
+        : IterType::Iteration;
+
+    auto orig_id =
+        info_->initialInfo()->getDynamicResizedIterDomains().at(id_index);
+
+    // id might have been updated in the previous loop
+    IterDomain* id = maybeMutated(orig_id)->as<IterDomain>();
+
     NVF_CHECK(
         id->definition() && id->definition()->isA<Resize>(),
         "Resized IterDomain must have a Resize definition");
     auto def = id->definition()->as<Resize>();
+
+    Val* left_expand_val = def->leftExpand();
+    if (!def->leftExpand()->isConst()) {
+      left_expand_val =
+          IrBuilder::create<Val>(left_expand, def->leftExpand()->dtype());
+      registerConcretization(def->leftExpand(), left_expand_val);
+    }
+
+    Val* right_expand_val = def->rightExpand();
+    if (!def->rightExpand()->isConst()) {
+      right_expand_val =
+          IrBuilder::create<Val>(right_expand, def->rightExpand()->dtype());
+      registerConcretization(def->rightExpand(), right_expand_val);
+    }
+
     auto new_id = IterDomain::resize(
-        def->in(),
-        def->leftExpand(),
-        def->rightExpand(),
+        maybeMutated(def->in())->as<IterDomain>(),
+        left_expand_val,
+        right_expand_val,
         id->isRFactorProduct(),
         iter_type);
 
-    registerConcretization(id, new_id);
+    registerConcretization(orig_id, new_id);
+
+    // Concretize the output shape which is constant
+    if (!orig_id->extent()->sameAs(new_id->extent())) {
+      registerConcretization(orig_id->extent(), new_id->extent());
+    }
```

Comment:
Why are we not handling expanded extent here?

i.e. doesn't slice an expanded dimension preserve the expanded field here, which means we need to register that in concretization as well?

Reply:
It's a good point that we should handle expanded extents on the _inputs_. Ideally we would need to handle output expanded extent so I will update the PR. However, note that currently I believe if you resize an expanded ID, the result is `Iteration`. This is because we do not check that the pad values are negative, so we might be doing an actual padding of that dimension. In case the pad values are constant we could do that check and sometimes preserve expanded broadcasts in slice.